### DISCUSSION TITLE
NMA with EM-data using pseudoatoms

### DISCUSSION THREAD

**Discussion by DavidHaselbach** [prody/ProDy#1881] [Archived Issues]
Hi, 

I am trying to do an NMA with an Electron-microscopy-map. Therefore I produce a pseudo-atom map (as pdb file) 
this looks like: 

REMARK xmipp_convert_vol2pseudo
REMARK fixedGaussian 8.000000
REMARK intensityColumn Bfactor
ATOM      1 DENS DENS    1     -11.600 -31.600   4.800     1  0.01      DENS
ATOM      2 DENS DENS    2     -43.600   8.400 -27.600     1  0.04      DENS
ATOM      3 DENS DENS    3     -14.400  20.400  16.400     1  0.01      DENS
ATOM      4 DENS DENS    4     -39.200   0.000-120.000     1  0.06      DENS
ATOM      5 DENS DENS    5     -35.600  -3.200 -32.000     1  0.05      DENS
ATOM      6 DENS DENS    6     -24.000  28.400   4.000     1  0.02      DENS
ATOM      7 DENS DENS    7      88.400  -2.800 -76.000     1  0.01      DENS
ATOM      8 DENS DENS    8     -40.000  16.000 -24.000     1  0.07      DENS
ATOM      9 DENS DENS    9      24.000 -43.600 -75.200     1  0.01      DENS
ATOM     10 DENS DENS   10      -4.000   0.000 -75.600     1  0.02      DENS 

Unfortunatly I can not parse that file. I get as error, that this is not a valid pdb file or identifier. Is it possiple to somehow work with those files? 

Cheers

David

**Comment by chakracs** [comment]
do the pseudo-atom maps (as PDB files) have a 'standard' format, like the one you show here?

**Comment by DavidHaselbach** [comment]
yes exactly. That is the standard format

**Comment by cihankayacihan** [comment]
We will add a parser for EM map standard format.

**Comment by jamesmkrieger** [comment]
The problem is probably that the residue name should be DEN not DENS because the PDB file format says you need 3 columns for that.

**Comment by jamesmkrieger** [comment]
Any update on this?

**Comment by jamesmkrieger** [comment]
We will look into this more at some point I guess.

**Comment by jamesmkrieger** [comment]
We now have our own tool within ProDy for performing ANM NMA using beads fitted to cryo-EM maps using the topology representing network algorithm. See http://prody.csb.pitt.edu/tutorials/cryoem_tutorial/

Nevertheless, it would be good to handle this output from XMIPP. @ajimoreno, maybe you can help with this.

**Comment by jamesmkrieger** [comment]
I found the relevant XMIPP code at https://github.com/I2PC/xmipp/blob/devel/src/xmipp/libraries/reconstruction/volume_to_pseudoatoms.cpp and it still has this format so we probably should provide support for it at some point.

**Comment by jamesmkrieger** [comment]
I think this is probably a niche format, which doesn't really conform to PDB format at all. The fields do not even start at the right columns. For example, field 2 starts at column 6 (starting from 1), which is supposed to be part of the first field so the first field is processed as "ATOM 1". We probably need a completely new parser for it and I don't think that's very far up our priority list right now. I plan to do more ProDy/XMIPP integration next year so probably this will be part of that.

**Comment by jamesmkrieger** [comment]
I think the way to fix that is on the xmipp side and I think I've found a way to do that

**Comment by jamesmkrieger** [comment]
There is a parser for this in scipion-em-continuousflex that comes from Situs ModeHunter that could be copied, but I think it's better to change xmipp and continuousflex instead.
